File: numbers/src/main/java/org/dicio/numbers/formatter/param/NiceDateParameters.java
package org.dicio.numbers.formatter.param;

import org.dicio.numbers.formatter.Formatter;

import java.time.LocalDate;

public class NiceDateParameters {
    // TODO documentation

    private final Formatter formatter;
    private final LocalDate date;

    // default values
    private LocalDate now = null;

    public NiceDateParameters(final Formatter formatter, final LocalDate date) {
        this.formatter = formatter;
        this.date = date;
    }

    /**
     *
     * @param now nullable
     * @return
     */
    public NiceDateParameters now(final LocalDate now) {
        this.now = now;
        return this;
    }

    public String get() {
        return formatter.niceDate(date, now);
    }
}

File: numbers/src/main/java/org/dicio/numbers/formatter/param/NiceDateParameters.kt
package org.dicio.numbers.formatter.param

import org.dicio.numbers.formatter.Formatter
import java.time.LocalDate

/**
 * TODO documentation
 */
class NiceDateParameters(
    private val formatter: Formatter, private val date: LocalDate
) {
    // default values
    private var now: LocalDate? = null

    fun now(now: LocalDate?): NiceDateParameters {
        this.now = now
        return this
    }

    fun get(): String {
        return formatter.niceDate(date, now)
    }
}
